src/scripts/upload-wpt-results.py

```diff
@@ -47,205 +43,85 @@ def main(raw_results_directory, platform_id, browser_name, browser_channel,
-        payload_str = json.dumps(payload)
-        f.write(payload_str)
+            data = json.load(handle)
 
+        assert 'run_info' in data
+        report['run_info'] = data['run_info']
 
-def summarize(filenames):
-    summary = {}
+        assert 'time_start' in data
+        if data['time_start'] < report['time_start']:
+            report['time_start'] = data['time_start']
 
-    for filename in filenames:
-        with open(filename) as handle:
-            data = json.load(handle)
+        assert 'time_end' in data
+        if data['time_end'] > report['time_end']:
+            report['time_end'] = data['time_end']
 
         assert 'results' in data
         assert isinstance(data['results'], list)
+        report['results'] += data['results']
 
-        for result in data['results']:
-            test_file = result['test']
-
-            assert test_file not in summary, (
-                'test_file "%s" is not already present in summary')
-
-            assert 'status' in result
-
-            if result['status'] in ('OK', 'PASS'):
-                summary[test_file] = [1, 1]
-            else:
-                summary[test_file] = [0, 1]
-
-            assert 'subtests' in result
-            assert isinstance(result['subtests'], list)
-
-            for subtest in result['subtests']:
-
-                assert 'status'in subtest
-
-                if subtest['status'] == 'PASS':
-                    summary[test_file][0] += 1
-
-                summary[test_file][1] += 1
-
-    return summary
-
-
-def expand_os_version(name, version):
-    '''Expand the string "*" to describe the version of the system running this
-    script.
-
-    This behavior assumes that the uploading script is run on the same platform
-    from which the results were collected. This assumption may not always hold,
-    but the functionality is implemented in order to satisfy the expectations
-    of the data's consumer.'''
-
-    if version != '*':
-        return version
-
-    if name != platform.system().lower():
-        raise ValueError('Cannot expand version wildcard for a foreign system')
-
-    match = re.search(r'[0-9]+\.[0-9]+', platform.release())
-
-    if match is None:
-        raise ValueError('Cannot infer host platform version')
-
-    return match.group(0)
-
-
-def notify(url, secret, payload):
-    response = requests.post(url,
-                             params={'secret': secret},
-                             data=json.dumps(payload))
+    # When the report is missing critical metadata, extend it with information
+    # provided via the command-line. (Currently, the WPT CLI does not include
+    # this metadata in reports generated via the Sauce Labs service.)
+    assert 'run_info' in report
+    assert isinstance(data['run_info'], object)
+    for name in (u'product', 'browser_version', 'os', 'os_version'):
+        if name not in report['run_info']:
+            report['run_info'][name] = platform_info[name]
```

Comment:
This might not be correct.

I suspect Sauce runs do have 'product' (or maybe even other fields), but the wrong values (e.g. product=sauce, os=[client os]). Can you get a sample wptreport from a Sauce run? (Doesn't need to be the consolidated version; each shard should also have `run_info`.)

Reply:
You're right. The metadata is present and incorrect. Here's the report generated for a test run in Apple Safari via Sauce Labs:

```json
{
    "results": [
        {
            "message": "/infrastructure/assumptions/ahem.html d4a5ec8885c67c8491d08fa7b04268aba6a19480\n/infrastructure/assumptions/ahem-ref.html ba37c5363f2183149d7267dec4446634846b4721\nTesting d4a5ec8885c67c8491d08fa7b04268aba6a19480 == ba37c5363f2183149d7267dec4446634846b4721",
            "status": "FAIL",
            "subtests": [],
            "test": "/infrastructure/assumptions/ahem.html"
        }
    ],
    "run_info": {
        "bits": 64,
        "debug": false,
        "has_sandbox": true,
        "linux_distro": "Ubuntu",
        "os": "linux",
        "os_version": "16.04",
        "processor": "x86_64",
        "product": "sauce",
        "revision": "8c8238ca96f913c1b90f838def7f9cf0253c0113",
        "version": "Ubuntu 16.04"
    },
    "time_end": 1527784288550,
    "time_start": 1527784271236
}
```

[You've already offered to change this in WPT](), but I'd be glad to do it as
well.

Know that the build process is already aware of whether this report comes from
a Sauce-Labs-mediated browser. That means we have the information necessary to
address this without an upstream change. We should improve the WPT CLI
regardless, but that might take some time. If there's urgency here, I could
get this working by the end of the day.

Let me know how you'd like to proceed

Reply:
Forgot the link for "You've already offered to change this in WPT": https://github.com/web-platform-tests/wpt.fyi/pull/208#discussion_r191932905

Reply:
@jugglinmike let's override the metadata for all Sauce runs and Sauce runs only (instead of filling in missing fields, which would also hide issues in Firefox/Chrome metadata that are supposed to be correct).